### ISSUE TITLE
Bad Credentials due to username?

### ISSUE THREAD

**Issue by chrfriese123** [python-webuntis/python-webuntis#52]
Hi all,

my problem is related to the Home Assistant integration of python-webuntis. I already placed it here: [https://github.com/JonasJoKuJonas/homeassistant-WebUntis/issues/89](https://github.com/JonasJoKuJonas/homeassistant-WebUntis/issues/89) But I'm not sure on which side the issue is.

**Short description**
I installed the integration and wanted to setup the credentials for two instances. The first one works fine, the second one for the same school doesn't work. Feedback is 
```
2023-10-11 20:56:45.517 DEBUG (SyncWorker_1) [webuntis] Making new request:
2023-10-11 20:56:45.517 DEBUG (SyncWorker_1) [webuntis] URL: https://tipo.webuntis.com/WebUntis/jsonrpc.do?school=name+city
2023-10-11 20:56:46.057 DEBUG (SyncWorker_1) [webuntis] Valid JSON found
2023-10-11 20:56:46.057 DEBUG (SyncWorker_1) [webuntis]   Got data{"jsonrpc":"2.0","id":"2023-10-11 20:56:45.517114","error":{"message":"bad credentials","code":-8504
2023-10-11 20:56:46.058 ERROR (SyncWorker_1) [webuntis] {'jsonrpc': '2.0', 'id': '2023-10-11 20:56:45.517114', 'error': {'message': 'bad credentials', 'code': -8504}}
```
I tried the suspicious credentials with a simple python script and get access as expected. The credentials are
```
server: https://tipo.webuntis.com
school: name+city
username: 7d
password: Krauß
```
**Observation**
The only difference I could spot is that my python-webuntis has version **0.1.22**, whereas the home assistant version has **0.1.21**. 

I assume that the German **ß** could be the source of my issue. 

Could the difference in versions be a root cause?

**Comment by chrfriese123** [comment]
Issue was solved with an update of the integration from 0.1.21 to 0.1.22
Maybe some setting change, but it works now.

**Comment by AugustH** [comment]
there was an issue with the password encoding in 0.1.21, so the update was the right choice 😁
